### PR TITLE
Factorize close

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Yet another part of https://github.com/ankidroid/Anki-Android/pull/11418 which can be considered separately

It considers the actions that can close preferences

### CODE REVIEW

**Comment by BrayanDSO** [comment]
`AppCompatPreferenceActivity.kt` will be gone when someone takes up #5019 and DeckOptions/FilteredDeckOptions probably will inherit from `AnkiActivity` then. Even so, these refactors may help in the future, so I'm just stating it.

**Comment by Arthur-Milchior** [comment]
Thanks.
FYI, @BrayanDSO , even if you can't merge a PR, you can review it. Since you seemed to already have an idea about it, you should be able to use github to approve it (or require change)

**Review by mikehardy — APPROVED** [review]
LGTM but needs de-conflicting

**Comment by Arthur-Milchior** [comment]
@mikehardy done

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt [hunk 1]
```diff
@@ -586,13 +584,10 @@ class DeckOptions :
...
-    override fun onOptionsItemSelected(item: MenuItem): Boolean {
-        if (item.itemId == android.R.id.home) {
-            closeWithResult()
-            return true
-        }
-        return false
+    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, key: String?) {
+        // update values on changed preference
+        prefChanged = true
+        updateSummaries()
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckOptions.kt:590]: This seems like it is from #11423 / doesn't fit the commit purpose

**Review by BrayanDSO — CHANGES_REQUESTED** [review]
I think that this may have been contaminated by one of the other related PRs. The rest looks good to me.

**Review by BrayanDSO — APPROVED** [review]
LGTM. Thanks for the changes. DeckOptions/FilteredDeckOptions aren't very pleasant files, so this cleaning helps a lot.

Waiting CI

**Pull request merged**